### ISSUE TITLE
Protocols with NULL read_wan priority can cause the server to crash

### ISSUE THREAD

**Issue by dchristidis** [rucio/rucio#4237]
Motivation
----------

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/rucio/web/rest/replica.py", line 420, in POST
    vo=ctx.env.get('vo')):
  File "/usr/local/lib/python3.6/site-packages/rucio/api/replica.py", line 186, in list_replicas
    for rep in replicas:
  File "/usr/local/lib/python3.6/site-packages/rucio/db/sqla/session.py", line 339, in new_funct
    for row in function(*args, **kwargs):
  File "/usr/local/lib/python3.6/site-packages/rucio/core/replica.py", line 1208, in list_replicas
    session):
  File "/usr/local/lib/python3.6/site-packages/rucio/core/replica.py", line 928, in _list_replicas
    rse_info[rse_id]['priority_wan'] = {p['scheme']: p['domains']['wan']['read'] for p in rse_info[rse_id]['protocols'] if p['domains']['wan']['read'] > 0}
  File "/usr/local/lib/python3.6/site-packages/rucio/core/replica.py", line 928, in <dictcomp>
    rse_info[rse_id]['priority_wan'] = {p['scheme']: p['domains']['wan']['read'] for p in rse_info[rse_id]['protocols'] if p['domains']['wan']['read'] > 0}
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```

Modification
------------

Consider either adding some protection against this in the core or adding a database constraint.

**Comment by dchristidis** [comment]
Also applies to `read_lan`.

**Comment by dchristidis** [comment]
I’m afraid the fix is inadequate because it does not cover activities that are not explicitly defined. For example, something like:

```console
$ rucio-admin rse add-protocol --hostname example.com --scheme mock --prefix / --port 0 --impl rucio.rse.protocols.mock.Default --domain-json '{"wan": {"delete": 1}}' RSE
```

**Comment by dchristidis** [comment]
Closed in favour of  #6201.
